unlock-app/src/components/content/event/Layout/PastEvent/index.tsx

```diff
@@ -0,0 +1,186 @@
+import { Event, PaywallConfigType } from '@unlock-protocol/core'
+import { MdAssignmentLate } from 'react-icons/md'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useHasClaimedRefund } from '~/hooks/useHasClaimedRefund'
+import { useMutation } from '@tanstack/react-query'
+import { ethers } from 'ethers'
+import { config } from '~/config/app'
+import KickbackAbi from '../../Settings/Components/Kickback/KickbackAbi'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { useGetApprovedRefunds } from '~/hooks/useGetApprovedRefunds'
+import { useMemo } from 'react'
+
+export const ClaimRefund = ({
+  refundProofAndValue,
+  lockAddress,
+  network,
+  refreshHasClaimedRefund,
+}: {
+  refreshHasClaimedRefund: () => void
+  refundProofAndValue: any
+  lockAddress: string
+  network: number
+}) => {
+  // Get the tree!
+  // Check if balance on lock is enough, show error if not!
+  // If they have not, show a button to claim a refund.
+  const { getWalletService } = useAuth()
+  const { kickbackAddress } = config.networks[network]
+
+  const claimRefund = useMutation(async () => {
+    const walletService = await getWalletService(network)
+    const contract = new ethers.Contract(
+      kickbackAddress!,
+      KickbackAbi,
+      walletService.signer
+    )
+
+    await ToastHelper.promise(
+      contract
+        .refund(
+          lockAddress,
+          refundProofAndValue.proof,
+          refundProofAndValue.leaf[1]
+        )
+        .then((tx: any) => tx.wait())
+        .then(() => refreshHasClaimedRefund()),
+      {
+        success: 'Your refund has been issued!',
+        error: 'We could not issue your refund. Please try again later.',
+        loading: `Issuing refund...`,
+      }
+    )
+  })
+
+  const claim = async () => {
+    await claimRefund.mutateAsync()
+  }
+
+  return (
+    <>
+      <p>You attended this event are your wallet is elligible for a refund!</p>
+      <Button onClick={claim} loading={claimRefund.isLoading}>
+        Claim Refund
+      </Button>
+    </>
+  )
+}
+
+export const ClaimRefundInfo = ({
+  event,
+  checkoutConfig,
+}: {
+  event: Event
+  checkoutConfig: {
+    id?: string
+    config: PaywallConfigType
+  }
+}) => {
+  const lockAddress = Object.keys(checkoutConfig.config.locks)[0]
+  const network = (checkoutConfig.config.locks[lockAddress].network ||
+    checkoutConfig.config.network)!
+
+  const { account } = useAuth()
+  const {
+    data: hasClaimedRefund,
+    isLoading: isLoadingHasClaimedRefund,
+    refetch: refreshHasClaimedRefund,
+  } = useHasClaimedRefund(lockAddress, network, account)
+
+  const { data: approvedRefunds } = useGetApprovedRefunds(event)
+
+  const refundProofAndValue = useMemo(() => {
+    if (!approvedRefunds || !account) {
+      return false
+    }
+    // @ts-expect-error Type 'any[] | undefined' is not assignable to type 'any[]'.
+    const tree = StandardMerkleTree.load(approvedRefunds)
```

Comment:
This definition is coming from Typescript... and I am not sure how to force typescript to not add the `| undefined` part :/